### PR TITLE
Fixed toggle Button contrast in Light Theme

### LINKED ISSUE #500: Frontend: Contrast Issue with Light/Dark Mode Toggle Button
## Bug description
I Believe there is some contrast issue with dark/light mode toggle button, It is bit difficult for a person visiting first time to identify the toggle button in light mode. It looks like a dot rather than a toggle switch ( when compared with its background color ).
Screenshot from 2021-12-12 23-00-49

## Expected behavior
Better visibility of the button with better contrast ratio

## How to reproduce

1.Run the application locally
2.Observe the toggle button in top left and change it to light mode
3.Observe the same button which is difficult to predict it as toggle switch

## Version information 
* **Signoz version**:
* **Browser version**: Brave Version 1.32.113 Chromium: 96.0.4664.45 (Official Build) (64-bit)
* **Your OS and version**: Ubuntu 20.04.3 LTS

### PR DESCRIPTION
**Pull request by Devesh21700Kumar**
Fixes #500 

Used isDarkMode from store and set backgroundColor on basis of theme (using a basic ternary operation)

Below are the SS:-

image

image

### CODE REVIEW

**Comment by Devesh21700Kumar** [comment]
@pal-sig  do check this out

**Comment by pal-sig** [comment]
Hi @Devesh21700Kumar can you please make it a styled component rather than using a style as a props

**Comment by Devesh21700Kumar** [comment]
> Hi @Devesh21700Kumar can you please make it a styled component rather than using a style as a props

Hey @pal-sig  actually I tried a lot but the conditional contrast set is not happening with styled components. This is because antd using props and styling its component using styled components was only leading to a single render.

You can try out my changes to verify as well

in style.ts

`export const ToggleButton = styled(Switch)<DarkModeProps>`
	background-color: ${({ checked }): string => (!checked ? 'red' : '')};
`;

interface DarkModeProps {
	checked: boolean;
	defaultChecked: boolean;
}`

`import { Menu, Typography } from 'antd';
import { ToggleButton } from './styles';`
and `<ToggleButton checked={isDarkMode} onChange={toggleTheme} defaultChecked={true} />`

in index.ts

**Comment by Devesh21700Kumar** [comment]
> > Hi @Devesh21700Kumar can you please make it a styled component rather than using a style as a props
> 
> Hey @pal-sig actually I tried a lot but the conditional contrast set is not happening with styled components. This is because antd using props and styling its component using styled components was only leading to a single render.
> 
> You can try out my changes to verify as well
> 
> in style.ts image
> 
> and
> 
> `import { Menu, Typography } from 'antd'; import { ToggleButton } from './styles';` and `<ToggleButton checked={isDarkMode} onChange={toggleTheme} defaultChecked={true} />`
> 
> in index.ts

I had tried using this and searched all on StackOverflow to avoid the issue but the conditional styling was not happening with antd styled componts.

@pal-sig do help

**Comment by pal-sig** [comment]
> > > Hi @Devesh21700Kumar can you please make it a styled component rather than using a style as a props
> > 
> > 
> > Hey @pal-sig actually I tried a lot but the conditional contrast set is not happening with styled components. This is because antd using props and styling its component using styled components was only leading to a single render.
> > You can try out my changes to verify as well
> > in style.ts image
> > and
> > `import { Menu, Typography } from 'antd'; import { ToggleButton } from './styles';` and `<ToggleButton checked={isDarkMode} onChange={toggleTheme} defaultChecked={true} />`
> > in index.ts
> 
> I had tried using this and searched all on StackOverflow to avoid the issue but the conditional styling was not happening with antd styled componts.
> 
> @pal-sig do help

already shared the context over slack

**Review by Devesh21700Kumar — COMMENTED** [review]
@pal-sig Thanks. It is refactored to styled components successfully now.

File: frontend/src/container/SideNav/index.tsx [hunk 1]
```diff
@@ -57,7 +58,7 @@ const SideNav = ({ toggleDarkMode }: Props): JSX.Element => {
 	return (
 		<Sider collapsible collapsed={collapsed} onCollapse={onCollapse} width={200}>
 			<ThemeSwitcherWrapper>
-				<ToggleButton checked={isDarkMode} onChange={toggleTheme} />
+				<ToggleButton checked={isDarkMode} onChange={toggleTheme} defaultChecked={true} />
```
Comment (pal-sig) [inline frontend/src/container/SideNav/index.tsx:61]: please check `defaultChecked` according to the darkMode

don't set the explicit value

File: frontend/src/container/SideNav/styles.ts [hunk 2]
```diff
@@ -24,3 +24,15 @@ export const Sider = styled(SiderComponent)`
...
 `;
+
+interface DarkModeProps {
+	checked?: boolean;
+	defaultChecked?: boolean;
+}
+
+export const ToggleButton = styled(Switch)<DarkModeProps>`
+	&&& {
+		background: ${({checked})=> (checked === false ? 'grey': ' ')};
```
Comment (pal-sig) [inline frontend/src/container/SideNav/styles.ts:35]: if checked is true 

can we use `unset` ?

**Review by pal-sig — CHANGES_REQUESTED** [review]

File: frontend/src/container/SideNav/styles.ts (hunk 2 above)
Comment (Devesh21700Kumar) [reply frontend/src/container/SideNav/styles.ts:35]: yes, as if it is true then default colour of antd component will show up

Comment (Devesh21700Kumar) [reply frontend/src/container/SideNav/styles.ts:35]: image

like this

File: frontend/src/container/SideNav/index.tsx (hunk 1 above)
Comment (Devesh21700Kumar) [reply frontend/src/container/SideNav/index.tsx:61]: Ok

Comment (Devesh21700Kumar) [reply frontend/src/container/SideNav/index.tsx]: @pal-sig  changed

**Review by pal-sig — APPROVED** [review]

File: frontend/src/container/SideNav/styles.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Devesh21700Kumar) [reply frontend/src/container/SideNav/styles.ts]: > can you use this
> 
> ```
> background: ${({ checked }) => checked === false && 'grey'};
> ```

@pal-sig done.

**Review by pal-sig — APPROVED** [review]

**Pull request merged**